File: 1894. Merge Strings Alternately/Solution.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        int i=0;
        int j=0;
        StringBuilder merged = new StringBuilder();     
        while(i<word1.length()&&j<word2.length()){
            merged.append(word1.charAt(i));
            merged.append(word2.charAt(j));
            i++;
            j++;
        }
        while(i<word1.length()){
            merged.append(word1.charAt(i));
            i++;
        }
        while(j<word2.length()){
            merged.append(word2.charAt(j));
            j++; 
        
        }   
        return merged.toString();
    }
}

File: 1894. Merge Strings Alternately/Solution.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        new=[]
        i=0
        while i<len(word1) or i<len(word2):
            if i<len(word1):
                new.append(word1[i])
            if i<len(word2):
                new.append(word2[i])
            i+=1
        return "".join(new)

        